--- org/CMakeLists.txt	2018-06-23 16:18:06.000000000 +0200
+++ mod/CMakeLists.txt	2018-06-23 22:26:05.631681268 +0200
@@ -668,7 +668,7 @@
 # OpenZWave
 # try to find open-zwave, if found, include support
 #
-option(USE_STATIC_OPENZWAVE "Build with static OpenZwave libraries" YES)
+option(USE_STATIC_OPENZWAVE "Build with static OpenZwave libraries" NO)
 if(USE_STATIC_OPENZWAVE)
   find_library(OpenZWave NAMES libopenzwave.a HINTS "../open-zwave-read-only" "../open-zwave-read-only/cpp/build")
   set(OPENZWAVE_LIB ${OpenZWave})
@@ -788,7 +788,7 @@
 
 SET(CPACK_SOURCE_PACKAGE_FILE_NAME "${domoticz}-dev")
 SET(CPACK_PACKAGING_INSTALL_PREFIX "/")
-SET(CPACK_PACKAGE_DEFAULT_LOCATION "/opt/${CPACK_PACKAGE_NAME}")
+SET(CPACK_PACKAGE_DEFAULT_LOCATION "/usr/share/${CPACK_PACKAGE_NAME}")
 SET(CPACK_PACKAGE_EXECUTABLES "domoticz;Home Automation System (Domotica).")
 
 #set(CPACK_DEB_COMPONENT_INSTALL TRUE)
@@ -797,7 +797,7 @@
 INCLUDE(CPack)
 
 IF(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
-  set(CMAKE_INSTALL_PREFIX "/opt/${CPACK_PACKAGE_NAME}" CACHE PATH "Install path prefix, prepended onto install directories." FORCE)
+  set(CMAKE_INSTALL_PREFIX "/usr/share/${CPACK_PACKAGE_NAME}" CACHE PATH "Install path prefix, prepended onto install directories." FORCE)
 ENDIF(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
 
 install(TARGETS domoticz DESTINATION ${CMAKE_INSTALL_PREFIX})
