--- org/hardware/hardwaretypes.h	2020-06-07 14:47:57.833369112 +0200
+++ mod/hardware/hardwaretypes.h	2020-06-07 15:34:20.840335508 +0200
@@ -409,6 +409,7 @@
 	uint8_t len;
 	uint8_t type;
 	uint8_t subtype;
+	uint8_t rssi;
 	uint8_t	id1;
 	uint8_t	id2;
 	uint8_t	id3;
@@ -422,6 +423,7 @@
 		ar & cereal::make_nvp("len", len);
 		ar & cereal::make_nvp("type", type);
 		ar & cereal::make_nvp("subtype", subtype);
+		ar & cereal::make_nvp("rssi", rssi);
 		ar & cereal::make_nvp("id1", id1);
 		ar & cereal::make_nvp("id2", id2);
 		ar & cereal::make_nvp("id3", id3);
@@ -490,6 +492,8 @@
 	uint8_t len;
 	uint8_t type;
 	uint8_t subtype;
+	uint8_t rssi;
+	uint8_t battery_level;
 	uint8_t id;
 	float floatval1;
 	float floatval2;
@@ -503,6 +507,8 @@
 		ar & cereal::make_nvp("len", len);
 		ar & cereal::make_nvp("type", type);
 		ar & cereal::make_nvp("subtype", subtype);
+		ar & cereal::make_nvp("rssi", rssi);
+		ar & cereal::make_nvp("battery_level", battery_level);
 		ar & cereal::make_nvp("id", id);
 		ar & cereal::make_nvp("floatval1", floatval1);
 		ar & cereal::make_nvp("floatval2", floatval2);
@@ -517,6 +523,8 @@
 		type = pTypeGeneral;
 		subtype = sTypeVisibility;
 		id = 0;
+		battery_level = 255;
+		rssi = 12;
 		floatval1 = 0;
 		floatval2 = 0;
 		intval1 = 0;
