--- org/updatedomo	2020-03-05 22:26:48.000000000 +0100
+++ mod/updatedomo	2020-03-08 20:55:14.308182370 +0100
@@ -1,24 +1,3 @@
 #!/bin/sh
 
-# This script can be used to upgrade/compile to the latest version
-# Please note that all modifications on the original files are lost
-
-echo "Updating to server revision..."
-git fetch --all
-git reset --hard origin/master
-
-cmake -DCMAKE_BUILD_TYPE=Release .
-if [ $? -ne 0 ]
-then
-		echo "CMake failed!";
-		exit 1
-fi
-make
-if [ $? -ne 0 ]
-then
-		echo "Compile failed!";
-		exit 1
-fi
-# Restart domoticz
-echo "Restarting Domoticz... (please standby...)"
-sudo service domoticz.sh restart
+echo "Update of Domoticz is done by yum update in ClearOS"
